string dataDir
string filename
---
bool success